contrib/mod_snmp/mod_snmp.c

```diff
@@ -3326,30 +3326,43 @@ static void snmp_exit_ev(const void *event_data, void *user_data) {
 
   ev_incr_value(SNMP_DB_DAEMON_F_CONN_COUNT, "daemon.connectionCount", -1);
 
-  if (session.disconnect_reason == PR_SESS_DISCONNECT_SESSION_INIT_FAILED) {
-    ev_incr_value(SNMP_DB_DAEMON_F_CONN_REFUSED_TOTAL,
-      "daemon.connectionRefusedTotal", 1);
+  switch (session.disconnect_reason) {
+    case PR_SESS_DISCONNECT_BANNED:
+    case PR_SESS_DISCONNECT_CONFIG_ACL:
+    case PR_SESS_DISCONNECT_MODULE_ACL:
+    case PR_SESS_DISCONNECT_SESSION_INIT_FAILED:
+      ev_incr_value(SNMP_DB_DAEMON_F_CONN_REFUSED_TOTAL,
+        "daemon.connectionRefusedTotal", 1);
+      break;
 
-  } else {
-    const char *proto;
+    case PR_SESS_DISCONNECT_SEGFAULT:
+      ev_incr_value(SNMP_DB_DAEMON_F_SEGFAULT_COUNT,
+        "daemon.segfaultCount", 1);
+      break;
```

Comment:
This case is needed for properly incrementing the `daemon.segfaultCount` counter.